csharp/src/Ice/InputStream.cs

```diff
@@ -592,456 +208,526 @@ public long ReadLong()
+        /// <returns>The enum value (int) read from the stream.</returns>
+        public int ReadEnumValue() => ReadSize();
+
+        /// <summary>Reads a sequence of fixed-size numeric type from the stream and returns an array.</summary>
+        /// <param name="elementSize">The size of each element of the sequence, in bytes. Must be sizeof(T).</param>
+        /// <returns>The sequence read from the stream, as an array.</returns>
+        public T[] ReadFixedSizeNumericArray<T>(int elementSize) where T : struct
         {
             Debug.Assert(_buffer.Array != null);
-            float value = BitConverter.ToSingle(_buffer.Array, _buffer.Offset + _pos);
-            _pos += 4;
+            T[] value = new T[ReadAndCheckSeqSize(elementSize)];
+            int byteCount = Buffer.ByteLength(value);
+            Debug.Assert(_buffer.Count - _pos >= byteCount);
+            Buffer.BlockCopy(_buffer.Array, _buffer.Offset + _pos, value, 0, byteCount);
+            _pos += byteCount;
             return value;
         }
 
-        /// <summary>
-        /// Extracts an optional float value from the stream.
-        /// </summary>
-        /// <param name="tag">The numeric tag associated with the value.</param>
-        /// <returns>The optional value.</returns>
-        public float? ReadFloat(int tag)
+        /// <summary>Reads a proxy from the stream.</summary>
+        /// <param name="factory">The proxy factory used to create the typed proxy.</param>
+        /// <returns>The proxy read from the stream.</returns>
+        public T? ReadProxy<T>(ProxyFactory<T> factory) where T : class, IObjectPrx
         {
-            if (ReadOptional(tag, OptionalFormat.F4))
+            var ident = new Identity(this);
+            if (ident.Name.Length == 0)
             {
-                return ReadFloat();
+                return null;
             }
             else
             {
-                return null;
+                return factory(Communicator.CreateReference(ident, this));
             }
         }
 
-        /// <summary>
-        /// Extracts a sequence of float values from the stream.
-        /// </summary>
-        /// <returns>The extracted float sequence.</returns>
-        public float[] ReadFloatArray()
-        {
-            float[] value = new float[ReadAndCheckSeqSize(4)];
-            ReadNumericArray(value);
-            return value;
-        }
+        /// <summary>Reads a sequence from the stream.</summary>
+        /// <param name="minElementSize">The minimum size of each element of the sequence, in bytes.</param>
+        /// <param name="reader">The input stream reader used to read each element of the sequence.</param>
+        /// <returns>The sequence read from the stream as an ICollection{T}. The return value does not fully implement
+        /// ICollection{T}, in particular you can only call GetEnumerator() once on this collection.</returns>
```

Comment:
I find the comment misleading, it doesn't return the sequence read from the stream (only the size was read at the point the methods return), the collection is later read upon enumeration and that is why `GetEnumerator` should only be called once.

Reply:
Fixed.